app/assets/graphing/graph.js

```diff
@@ -120,7 +120,19 @@ document.addEventListener("DOMContentLoaded", function(event) {
                 if (type === "bar") {
                     var bars = d3.selectAll(".c3-bars-" + yID + " path")[0];
                     for (var i = 0; i < bars.length; i++) {
-                        bars[i].style.opacity = lineOpacities[yID];
+                        // If there's a bar-specific color, set it
+                        if (barColors[yID] && barColors[yID][i]) {
+                            bars[i].style.fill = barColors[yID][i];

```

Comment:
C3 actually supports bar-specific colors, but it's easier to just handle the colors here along with the opacities (which are not supported at all by C3).